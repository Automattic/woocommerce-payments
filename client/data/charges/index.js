/** @format */

/**
 * Internal dependencies
 */
import reducer from './reducer.gen';
import * as selectors from './selectors.gen';
import * as actions from './actions.gen';
import * as resolvers from './resolvers';

export { reducer, selectors, actions, resolvers };
export * from './hooks.gen';
