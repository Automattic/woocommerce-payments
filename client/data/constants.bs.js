// Generated by BUCKLESCRIPT, PLEASE EDIT WITH CARE


var namespace = "/wc/v3/payments";

var storeName = "wc/payments";

export {
  namespace ,
  storeName ,
  
}
/* No side effect */
