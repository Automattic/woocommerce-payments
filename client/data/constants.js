/** @format */

export const NAMESPACE = '/wc/v3/payments';
export const STORE_NAME = 'wc/payments';

export const ID_PREFIX = {
	transactions: 'transactions',
};
