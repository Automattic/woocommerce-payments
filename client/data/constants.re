let namespace = "/wc/v3/payments";
let storeName = "wc/payments";
