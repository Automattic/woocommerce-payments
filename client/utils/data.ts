/** @format */

export const getResourceId = (
	identifier: Record< string, string >
): string => {
	return JSON.stringify( identifier, Object.keys( identifier ).sort() );
};
