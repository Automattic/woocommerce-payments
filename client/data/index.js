/** @format */

/**
 * Internal dependencies
 */
import { STORE_NAME } from './constants';
import { initStore } from './store';

initStore();

export const WCPAY_STORE_NAME = STORE_NAME;

// We only ask for hooks when importing directly from 'data'.
export * from './deposits/hooks';
export * from './transactions/hooks';
export * from './charges/hooks';
export * from './timeline/hooks';
export * from './disputes/hooks';
export * from './settings/hooks';
export * from './multi-currency';
