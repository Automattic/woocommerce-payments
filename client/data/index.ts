/** @format */

/**
 * Internal dependencies
 */
import { STORE_NAME } from './constants';
import { initStore } from './store';

initStore();

// eslint-disable-next-line @typescript-eslint/naming-convention
export const WCPAY_STORE_NAME = STORE_NAME;

// We only ask for hooks when importing directly from 'wcpay/data'.
export * from './deposits/hooks';
export * from './transactions/hooks';
export * from './charges/hooks';
export * from './timeline/hooks';
export * from './disputes/hooks';
export * from './settings/hooks';
export * from './multi-currency';
export * from './card-readers/hooks';
export * from './capital/hooks';
export * from './documents/hooks';
export * from './payment-intents/hooks';
export * from './authorizations/hooks';
