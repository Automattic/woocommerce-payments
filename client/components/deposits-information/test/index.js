/**
 * External dependencies
 */
import React from 'react';
import { render } from '@testing-library/react';
import { merge } from 'lodash';

/**
 * Internal dependencies
 */
import DepositsInformation from '..';
import { useAllDepositsOverviews, useInstantDeposit } from 'wcpay/data';

jest.mock( 'wcpay/data', () => ( {
	useAllDepositsOverviews: jest.fn(),
	useInstantDeposit: jest.fn(),
} ) );

const createMockAccount = ( account = {} ) =>
	merge(
		{
			default_currency: 'eur',
			deposits_disabled: false,
			deposits_schedule: {
				delay_days: 7,
				interval: 'weekly',
				weekly_anchor: 'thursday',
			},
		},
		account
	);

const createMockCurrency = ( currencyCode, extra = {} ) =>
	merge(
		{
			currency: currencyCode,
			lastPaid: {
				id: 'po_...',
				date: 1619395200000,
				amount: 3160,
			},
			nextScheduled: {
				id: 'wcpay_estimated_weekly_eur_1622678400',
				date: 1622678400000,
				amount: 3343,
			},
			pending: {
				amount: 3343,
				deposits_count: 2,
			},
			available: {
				amount: 2030,
			},
		},
		extra
	);

const mockOverviews = ( currencies = null, account = null ) => {
	return useAllDepositsOverviews.mockReturnValue( {
		overviews: {
			currencies: currencies,
			account: account,
		},
		isLoading: null === currencies || ! currencies.length,
	} );
};

useInstantDeposit.mockReturnValue( {
	deposit: undefined,
	isLoading: false,
	submit: () => {},
} );

describe( 'Deposits information', () => {
	beforeEach( () => {
		global.wcpaySettings = { zeroDecimalCurrencies: [] };
	} );
	afterEach( () => {
		jest.clearAllMocks();
	} );

	test( 'renders correctly when loading', () => {
		mockOverviews();
		const { container } = render( <DepositsInformation /> );
		expect( container ).toMatchSnapshot();
	} );

	test( 'renders correctly with multiple currencies', () => {
		mockOverviews(
			[ createMockCurrency( 'usd' ), createMockCurrency( 'eur' ) ],
			createMockAccount()
		);

		const { container } = render( <DepositsInformation /> );
		expect( container ).toMatchSnapshot();
	} );

	test( 'renders correctly with all possible missing details', async () => {
		// Everything should be zeros, without extra details or errors.
		mockOverviews(
			[
				createMockCurrency( 'usd', {
					pending: null,
					nextScheduled: null,
					lastPaid: null,
					available: null,
				} ),
			],
			createMockAccount()
		);

		const { findAllByText, findAllByTestId } = render(
			<DepositsInformation />
		);

		expect( await findAllByText( '$0.00' ) ).toHaveLength( 4 );
		( await findAllByTestId( 'extra' ) ).forEach( ( extra ) => {
			expect( extra ).toBeEmptyDOMElement();
		} );
	} );

	test( 'renders instant deposit button only where applicable', async () => {
		const currencyWithInstantDeposit = createMockCurrency( 'usd', {
			instant: {
				amount: 12345,
			},
		} );

		const currencyWithoutInstantDeposit = createMockCurrency( 'eur' );

		mockOverviews(
			// Only one of the currencies in the snapshot should include the instant deposit button.
			[ currencyWithInstantDeposit, currencyWithoutInstantDeposit ],
			createMockAccount()
		);

		const { container, findByText } = render( <DepositsInformation /> );
		expect( await findByText( 'Instant deposit' ) ).toBeVisible();
		expect( container ).toMatchSnapshot();
	} );
} );
