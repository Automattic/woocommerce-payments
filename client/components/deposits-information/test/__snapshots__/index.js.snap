// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deposits information renders correctly when loading 1`] = `
<div>
  <div
    class="components-card is-size-medium css-1xs3c37-CardUI e1q7k77g0"
  >
    <div
      class="components-flex components-card__header is-size-small wcpay-deposits-information-header e1q7k77g1 css-1ufeymn-Flex-HeaderUI eboqfv50"
    >
      <div
        class="wcpay-deposits-information-header__heading"
      >
        <h3
          class="wcpay-deposits-information-header__title"
        >
          Deposits overview
        </h3>
        <p
          class="wcpay-deposits-information-header__schedule"
        >
          <svg
            class="gridicon gridicons-calendar wcpay-deposits-information-header__icon"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M19 4h-1V2h-2v2H8V2H6v2H5c-1.105 0-2 .896-2 2v13c0 1.104.895 2 2 2h14c1.104 0 2-.896 2-2V6c0-1.104-.896-2-2-2zm0 15H5V8h14v11z"
              />
            </g>
          </svg>
          <span
            aria-busy="true"
            class="is-loadable-placeholder is-inline"
          >
            Deposit schedule here
          </span>
        </p>
      </div>
    </div>
    <div
      class="components-card__body is-size-medium css-xmjzce-BodyUI e1q7k77g3"
    >
      <span
        aria-busy="true"
        class="is-loadable-placeholder is-inline"
      >
        Deposit information placeholder
      </span>
    </div>
  </div>
</div>
`;

exports[`Deposits information renders correctly with all possible missing details 1`] = `
<div>
  <div
    class="components-card is-size-medium css-1xs3c37-CardUI e1q7k77g0"
  >
    <div
      class="components-flex components-card__header is-size-small wcpay-deposits-information-header e1q7k77g1 css-1ufeymn-Flex-HeaderUI eboqfv50"
    >
      <div
        class="wcpay-deposits-information-header__heading"
      >
        <h3
          class="wcpay-deposits-information-header__title"
        >
          United States (US) dollar balance
        </h3>
        <p
          class="wcpay-deposits-information-header__schedule"
        >
          <svg
            class="gridicon gridicons-calendar wcpay-deposits-information-header__icon"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M19 4h-1V2h-2v2H8V2H6v2H5c-1.105 0-2 .896-2 2v13c0 1.104.895 2 2 2h14c1.104 0 2-.896 2-2V6c0-1.104-.896-2-2-2zm0 15H5V8h14v11z"
              />
            </g>
          </svg>
          Deposit schedule:
           
          Automatic, every business day â€“ your first deposit is held for seven days (
          <a
            href="https://docs.woocommerce.com/document/payments/faq/deposit-schedule/"
            rel="noopener noreferrer"
            target="_blank"
          >
            learn more
          </a>
          )
        </p>
      </div>
    </div>
    <div
      class="components-flex wcpay-deposits-information-row css-bz4r5k-Flex eboqfv50"
    >
      <div
        class="components-flex__block wcpay-deposits-information-block css-yr442k-Item-Block eboqfv52"
      >
        <div
          class="wcpay-deposits-information-block__title"
        >
          Pending balance
        </div>
        <div
          class="wcpay-deposits-information-block__value"
        >
          $0.00
        </div>
        <div
          class="wcpay-deposits-information-block__extra"
        />
      </div>
      <div
        class="components-flex__block wcpay-deposits-information-block css-yr442k-Item-Block eboqfv52"
      >
        <div
          class="wcpay-deposits-information-block__title"
        >
          Next deposit
        </div>
        <div
          class="wcpay-deposits-information-block__value"
        >
          $0.00
        </div>
        <div
          class="wcpay-deposits-information-block__extra"
        />
      </div>
    </div>
    <div
      class="components-flex wcpay-deposits-information-row css-bz4r5k-Flex eboqfv50"
    >
      <div
        class="components-flex__block wcpay-deposits-information-block css-yr442k-Item-Block eboqfv52"
      >
        <div
          class="wcpay-deposits-information-block__title"
        >
          Last deposit
        </div>
        <div
          class="wcpay-deposits-information-block__value"
        >
          $0.00
        </div>
        <div
          class="wcpay-deposits-information-block__extra"
        />
      </div>
      <div
        class="components-flex__block wcpay-deposits-information-block css-yr442k-Item-Block eboqfv52"
      >
        <div
          class="wcpay-deposits-information-block__title"
        >
          Available balance
        </div>
        <div
          class="wcpay-deposits-information-block__value"
        >
          $0.00
        </div>
        <div
          class="wcpay-deposits-information-block__extra"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Deposits information renders correctly with multiple currencies 1`] = `
<div>
  <div
    class="components-card is-size-medium css-1xs3c37-CardUI e1q7k77g0"
  >
    <div
      class="components-flex components-card__header is-size-small wcpay-deposits-information-header e1q7k77g1 css-1ufeymn-Flex-HeaderUI eboqfv50"
    >
      <div
        class="wcpay-deposits-information-header__heading"
      >
        <h3
          class="wcpay-deposits-information-header__title"
        >
          United States (US) dollar balance
        </h3>
        <p
          class="wcpay-deposits-information-header__schedule"
        >
          <svg
            class="gridicon gridicons-calendar wcpay-deposits-information-header__icon"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M19 4h-1V2h-2v2H8V2H6v2H5c-1.105 0-2 .896-2 2v13c0 1.104.895 2 2 2h14c1.104 0 2-.896 2-2V6c0-1.104-.896-2-2-2zm0 15H5V8h14v11z"
              />
            </g>
          </svg>
          Deposit schedule:
           
          Automatic, every week on Thursday
        </p>
      </div>
    </div>
    <div
      class="components-flex wcpay-deposits-information-row css-bz4r5k-Flex eboqfv50"
    >
      <div
        class="components-flex__block wcpay-deposits-information-block css-yr442k-Item-Block eboqfv52"
      >
        <div
          class="wcpay-deposits-information-block__title"
        >
          Pending balance
        </div>
        <div
          class="wcpay-deposits-information-block__value"
        >
          $33.43
        </div>
        <div
          class="wcpay-deposits-information-block__extra"
        >
          <a
            href="admin.php?page=wc-admin&path=%2Fpayments%2Fdeposits%2Fdetails&id=wcpay_estimated_weekly_eur_1622678400"
          >
            2 deposits
          </a>
        </div>
      </div>
      <div
        class="components-flex__block wcpay-deposits-information-block css-yr442k-Item-Block eboqfv52"
      >
        <div
          class="wcpay-deposits-information-block__title"
        >
          Next deposit
        </div>
        <div
          class="wcpay-deposits-information-block__value"
        >
          $33.43
        </div>
        <div
          class="wcpay-deposits-information-block__extra"
        >
          <a
            href="admin.php?page=wc-admin&path=%2Fpayments%2Fdeposits%2Fdetails&id=wcpay_estimated_weekly_eur_1622678400"
          >
            Est. Jun 3, 2021
          </a>
        </div>
      </div>
    </div>
    <div
      class="components-flex wcpay-deposits-information-row css-bz4r5k-Flex eboqfv50"
    >
      <div
        class="components-flex__block wcpay-deposits-information-block css-yr442k-Item-Block eboqfv52"
      >
        <div
          class="wcpay-deposits-information-block__title"
        >
          Last deposit
        </div>
        <div
          class="wcpay-deposits-information-block__value"
        >
          $31.60
        </div>
        <div
          class="wcpay-deposits-information-block__extra"
        >
          <a
            href="admin.php?page=wc-admin&path=%2Fpayments%2Fdeposits%2Fdetails&id=po_..."
          >
            April 26, 2021
          </a>
        </div>
      </div>
      <div
        class="components-flex__block wcpay-deposits-information-block css-yr442k-Item-Block eboqfv52"
      >
        <div
          class="wcpay-deposits-information-block__title"
        >
          Available balance
        </div>
        <div
          class="wcpay-deposits-information-block__value"
        >
          $20.30
        </div>
        <div
          class="wcpay-deposits-information-block__extra"
        />
      </div>
    </div>
  </div>
  <div
    class="components-card is-size-medium css-1xs3c37-CardUI e1q7k77g0"
  >
    <div
      class="components-flex components-card__header is-size-small wcpay-deposits-information-header e1q7k77g1 css-1ufeymn-Flex-HeaderUI eboqfv50"
    >
      <div
        class="wcpay-deposits-information-header__heading"
      >
        <h3
          class="wcpay-deposits-information-header__title"
        >
          Euro balance
        </h3>
        <p
          class="wcpay-deposits-information-header__schedule"
        >
          <svg
            class="gridicon gridicons-calendar wcpay-deposits-information-header__icon"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M19 4h-1V2h-2v2H8V2H6v2H5c-1.105 0-2 .896-2 2v13c0 1.104.895 2 2 2h14c1.104 0 2-.896 2-2V6c0-1.104-.896-2-2-2zm0 15H5V8h14v11z"
              />
            </g>
          </svg>
          Deposit schedule:
           
          Automatic, every week on Thursday
        </p>
      </div>
    </div>
    <div
      class="components-flex wcpay-deposits-information-row css-bz4r5k-Flex eboqfv50"
    >
      <div
        class="components-flex__block wcpay-deposits-information-block css-yr442k-Item-Block eboqfv52"
      >
        <div
          class="wcpay-deposits-information-block__title"
        >
          Pending balance
        </div>
        <div
          class="wcpay-deposits-information-block__value"
        >
          â‚¬33,43
        </div>
        <div
          class="wcpay-deposits-information-block__extra"
        >
          <a
            href="admin.php?page=wc-admin&path=%2Fpayments%2Fdeposits%2Fdetails&id=wcpay_estimated_weekly_eur_1622678400"
          >
            2 deposits
          </a>
        </div>
      </div>
      <div
        class="components-flex__block wcpay-deposits-information-block css-yr442k-Item-Block eboqfv52"
      >
        <div
          class="wcpay-deposits-information-block__title"
        >
          Next deposit
        </div>
        <div
          class="wcpay-deposits-information-block__value"
        >
          â‚¬33,43
        </div>
        <div
          class="wcpay-deposits-information-block__extra"
        >
          <a
            href="admin.php?page=wc-admin&path=%2Fpayments%2Fdeposits%2Fdetails&id=wcpay_estimated_weekly_eur_1622678400"
          >
            Est. Jun 3, 2021
          </a>
        </div>
      </div>
    </div>
    <div
      class="components-flex wcpay-deposits-information-row css-bz4r5k-Flex eboqfv50"
    >
      <div
        class="components-flex__block wcpay-deposits-information-block css-yr442k-Item-Block eboqfv52"
      >
        <div
          class="wcpay-deposits-information-block__title"
        >
          Last deposit
        </div>
        <div
          class="wcpay-deposits-information-block__value"
        >
          â‚¬31,60
        </div>
        <div
          class="wcpay-deposits-information-block__extra"
        >
          <a
            href="admin.php?page=wc-admin&path=%2Fpayments%2Fdeposits%2Fdetails&id=po_..."
          >
            April 26, 2021
          </a>
        </div>
      </div>
      <div
        class="components-flex__block wcpay-deposits-information-block css-yr442k-Item-Block eboqfv52"
      >
        <div
          class="wcpay-deposits-information-block__title"
        >
          Available balance
        </div>
        <div
          class="wcpay-deposits-information-block__value"
        >
          â‚¬20,30
        </div>
        <div
          class="wcpay-deposits-information-block__extra"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Deposits information renders instant deposit button only where applicable 1`] = `
<div>
  <div
    class="components-card is-size-medium css-1xs3c37-CardUI e1q7k77g0"
  >
    <div
      class="components-flex components-card__header is-size-small wcpay-deposits-information-header e1q7k77g1 css-1ufeymn-Flex-HeaderUI eboqfv50"
    >
      <div
        class="wcpay-deposits-information-header__heading"
      >
        <h3
          class="wcpay-deposits-information-header__title"
        >
          United States (US) dollar balance
        </h3>
        <p
          class="wcpay-deposits-information-header__schedule"
        >
          <svg
            class="gridicon gridicons-calendar wcpay-deposits-information-header__icon"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M19 4h-1V2h-2v2H8V2H6v2H5c-1.105 0-2 .896-2 2v13c0 1.104.895 2 2 2h14c1.104 0 2-.896 2-2V6c0-1.104-.896-2-2-2zm0 15H5V8h14v11z"
              />
            </g>
          </svg>
          Deposit schedule:
           
          Automatic, every week on Thursday
        </p>
      </div>
      <button
        class="components-button is-secondary"
        type="button"
      >
        Instant deposit
      </button>
    </div>
    <div
      class="components-flex wcpay-deposits-information-row css-bz4r5k-Flex eboqfv50"
    >
      <div
        class="components-flex__block wcpay-deposits-information-block css-yr442k-Item-Block eboqfv52"
      >
        <div
          class="wcpay-deposits-information-block__title"
        >
          Pending balance
        </div>
        <div
          class="wcpay-deposits-information-block__value"
        >
          $33.43
        </div>
        <div
          class="wcpay-deposits-information-block__extra"
        >
          <a
            href="admin.php?page=wc-admin&path=%2Fpayments%2Fdeposits%2Fdetails&id=wcpay_estimated_weekly_eur_1622678400"
          >
            2 deposits
          </a>
        </div>
      </div>
      <div
        class="components-flex__block wcpay-deposits-information-block css-yr442k-Item-Block eboqfv52"
      >
        <div
          class="wcpay-deposits-information-block__title"
        >
          Next deposit
        </div>
        <div
          class="wcpay-deposits-information-block__value"
        >
          $33.43
        </div>
        <div
          class="wcpay-deposits-information-block__extra"
        >
          <a
            href="admin.php?page=wc-admin&path=%2Fpayments%2Fdeposits%2Fdetails&id=wcpay_estimated_weekly_eur_1622678400"
          >
            Est. Jun 3, 2021
          </a>
        </div>
      </div>
    </div>
    <div
      class="components-flex wcpay-deposits-information-row css-bz4r5k-Flex eboqfv50"
    >
      <div
        class="components-flex__block wcpay-deposits-information-block css-yr442k-Item-Block eboqfv52"
      >
        <div
          class="wcpay-deposits-information-block__title"
        >
          Last deposit
        </div>
        <div
          class="wcpay-deposits-information-block__value"
        >
          $31.60
        </div>
        <div
          class="wcpay-deposits-information-block__extra"
        >
          <a
            href="admin.php?page=wc-admin&path=%2Fpayments%2Fdeposits%2Fdetails&id=po_..."
          >
            April 26, 2021
          </a>
        </div>
      </div>
      <div
        class="components-flex__block wcpay-deposits-information-block css-yr442k-Item-Block eboqfv52"
      >
        <div
          class="wcpay-deposits-information-block__title"
        >
          Available balance
        </div>
        <div
          class="wcpay-deposits-information-block__value"
        >
          $20.30
        </div>
        <div
          class="wcpay-deposits-information-block__extra"
        />
      </div>
    </div>
  </div>
  <div
    class="components-card is-size-medium css-1xs3c37-CardUI e1q7k77g0"
  >
    <div
      class="components-flex components-card__header is-size-small wcpay-deposits-information-header e1q7k77g1 css-1ufeymn-Flex-HeaderUI eboqfv50"
    >
      <div
        class="wcpay-deposits-information-header__heading"
      >
        <h3
          class="wcpay-deposits-information-header__title"
        >
          Euro balance
        </h3>
        <p
          class="wcpay-deposits-information-header__schedule"
        >
          <svg
            class="gridicon gridicons-calendar wcpay-deposits-information-header__icon"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M19 4h-1V2h-2v2H8V2H6v2H5c-1.105 0-2 .896-2 2v13c0 1.104.895 2 2 2h14c1.104 0 2-.896 2-2V6c0-1.104-.896-2-2-2zm0 15H5V8h14v11z"
              />
            </g>
          </svg>
          Deposit schedule:
           
          Automatic, every week on Thursday
        </p>
      </div>
    </div>
    <div
      class="components-flex wcpay-deposits-information-row css-bz4r5k-Flex eboqfv50"
    >
      <div
        class="components-flex__block wcpay-deposits-information-block css-yr442k-Item-Block eboqfv52"
      >
        <div
          class="wcpay-deposits-information-block__title"
        >
          Pending balance
        </div>
        <div
          class="wcpay-deposits-information-block__value"
        >
          â‚¬33,43
        </div>
        <div
          class="wcpay-deposits-information-block__extra"
        >
          <a
            href="admin.php?page=wc-admin&path=%2Fpayments%2Fdeposits%2Fdetails&id=wcpay_estimated_weekly_eur_1622678400"
          >
            2 deposits
          </a>
        </div>
      </div>
      <div
        class="components-flex__block wcpay-deposits-information-block css-yr442k-Item-Block eboqfv52"
      >
        <div
          class="wcpay-deposits-information-block__title"
        >
          Next deposit
        </div>
        <div
          class="wcpay-deposits-information-block__value"
        >
          â‚¬33,43
        </div>
        <div
          class="wcpay-deposits-information-block__extra"
        >
          <a
            href="admin.php?page=wc-admin&path=%2Fpayments%2Fdeposits%2Fdetails&id=wcpay_estimated_weekly_eur_1622678400"
          >
            Est. Jun 3, 2021
          </a>
        </div>
      </div>
    </div>
    <div
      class="components-flex wcpay-deposits-information-row css-bz4r5k-Flex eboqfv50"
    >
      <div
        class="components-flex__block wcpay-deposits-information-block css-yr442k-Item-Block eboqfv52"
      >
        <div
          class="wcpay-deposits-information-block__title"
        >
          Last deposit
        </div>
        <div
          class="wcpay-deposits-information-block__value"
        >
          â‚¬31,60
        </div>
        <div
          class="wcpay-deposits-information-block__extra"
        >
          <a
            href="admin.php?page=wc-admin&path=%2Fpayments%2Fdeposits%2Fdetails&id=po_..."
          >
            April 26, 2021
          </a>
        </div>
      </div>
      <div
        class="components-flex__block wcpay-deposits-information-block css-yr442k-Item-Block eboqfv52"
      >
        <div
          class="wcpay-deposits-information-block__title"
        >
          Available balance
        </div>
        <div
          class="wcpay-deposits-information-block__value"
        >
          â‚¬20,30
        </div>
        <div
          class="wcpay-deposits-information-block__extra"
        />
      </div>
    </div>
  </div>
</div>
`;
